addons/mail/models/mail_alias.py

```diff
@@ -105,17 +105,16 @@ def create(self, vals):
         if model_name:
             model = self.env['ir.model']._get(model_name)
             vals['alias_model_id'] = model.id
+            try:
+                self.env[model_name].check_access_rights('create')
+                self = self.sudo()
+            except AccessError:
+                pass
         if parent_model_name:
             model = self.env['ir.model']._get(parent_model_name)
             vals['alias_parent_model_id'] = model.id
         return super(Alias, self).create(vals)
 
-    @api.model
-    def _create_parent_record(self, parent_vals, model_name):
```

Comment:
@tde-banana-odoo what do you think of this alternative patch to avoid changing models.py?

Reply:
Seems already better indeed! I would move the self = self.sudo() in the else clause to try/except only the check access rights.